src/System.Linq/src/System/Linq/Select.cs

```diff
@@ -170,9 +172,58 @@ public int GetCount(bool onlyIfCheap)
             {
                 return _source.Length;
             }
+
+            public IPartition<TResult> Skip(int count)
+            {
+                if (count == 0) return (IPartition<TResult>)Clone();
+                if (count >= _source.Length) return new EmptyPartition<TResult>();
+                return new SelectIPartitionIterator<TSource, TResult>(new ListPartition<TSource>(_source, count, int.MaxValue), _selector);
+            }
+
+            public IPartition<TResult> Take(int count)
+            {
+                return count >= _source.Length ? (IPartition<TResult>)Clone() : new SelectIPartitionIterator<TSource, TResult>(new ListPartition<TSource>(_source, 0, count - 1), _selector);
+            }
```

Comment:
`arrayOfThousandElements.Select(i => i.ToString()).Skip(250).Take(250)` can be enumerated through in half the time. The savings can be very large with some queries.

Also with many uses this object would be available for collection immediately, if it was created as part of the query rather than held onto separately, which reduces the impact of the allocation. Previously the take iterator would have held onto a reference.